src/convert.jl

```diff
@@ -1,43 +1,38 @@
-#################### Unwrapping PropertyValues  ####################
-
-convert(::Type{BigInt}, pv::pm_perl_PropertyValue) = convert(BigInt, to_pm_Integer(pv))
-
-convert(::Type{<:Rational}, pv::pm_perl_PropertyValue) = convert(Rational{BigInt}, to_pm_Rational(pv))
-
-convert(::Type{Matrix{Rational{BigInt}}}, pv::pm_perl_PropertyValue) =
-    convert(Matrix{Rational{BigInt}}, to_matrix_rational(pv))
-
-convert(::Type{Matrix{BigInt}}, matrix::pm_perl_PropertyValue) =
-    convert(Matrix{BigInt}, to_matrix_int(pv))
+####################  Converting to polymake types  ####################
 
-convert(::Type{Vector{Rational{BigInt}}}, pv::pm_perl_PropertyValue) =
-    convert(Matrix{Rational{BigInt}}, to_vector_rational(pv))
+for (pm_T, jl_T) in [
+        (pm_Vector, AbstractVector),
+        (pm_Matrix, AbstractMatrix),
+        (pm_Array, AbstractVector),
+        (pm_Set, AbstractSet)
+        ]
+    @eval begin
+        convert(::Type{$pm_T}, itr::$jl_T) = $pm_T(itr)
+        convert(::Type{$pm_T{T}}, itr::$jl_T) where T = $pm_T{T}(itr)
+        convert(::Type{$pm_T}, itr::$pm_T) where T = itr
+        convert(::Type{$pm_T{T}}, itr::$pm_T) where T = itr
+    end
+end
 
-convert(::Type{Vector{BigInt}}, pv::pm_perl_PropertyValue) =
-    convert(Matrix{BigInt}, to_vector_int(pv))
+convert(::Type{pm_Set{T}}, itr::AbstractArray) where T = pm_Set{T}(itr)
 
-####################  Converting to polymake types  ####################
+###########  Converting to objects polymake understands  ###############
 
-convert(::Type{pm_Vector}, vec::AbstractVector) = pm_Vector(vec)
-convert(::Type{pm_Vector{T}}, vec::AbstractVector) where T = pm_Vector{T}(vec)
+struct PolymakeType end
 
-convert(::Type{pm_Matrix}, mat::AbstractMatrix) = pm_Matrix(mat)
-convert(::Type{pm_Matrix{T}}, mat::AbstractMatrix) where T = pm_Matrix{T}(mat)
-
-convert(::Type{pm_Array}, vec::AbstractVector) = pm_Array(vec)
-convert(::Type{pm_Array{T}}, vec::AbstractVector) where T = pm_Array{T}(vec)
+convert(::Type{PolymakeType}, x::T) where T = convert(convert_to_pm_type(T), x)
+convert(::Type{PolymakeType}, v::Visual) = v.obj
+convert(::Type{pm_perl_OptionSet}, dict) = pm_perl_OptionSet(dict)
 
-convert(::Type{pm_Set}, itr) = pm_Set(itr)
-convert(::Type{pm_Set{T}}, itr) where T = pm_Set{T}(itr)
-convert(::Type{pm_Set{T}}, as::AbstractSet) where T = pm_Set{T}(as)
+####################  Guessing the polymake type  ######################
 
-# disambiguations:
-convert(::Type{pm_Set}, as::AbstractSet) = pm_Set(as)
-convert(::Type{pm_Set{T}}, s::pm_Set{T}) where T = s
+# By default we throw an error:
+convert_to_pm_type(T::Type) = throw(ArgumentError("Unrecognized argument type: $T.\nYou need to convert to polymake compatible type first."))
```

Comment:
Is this intended to be a public function?

Reply:
no, external users should overload `convert(::Type{PolymakeType}, x::T) where T = ...` where the right hand side should consist of `pm_types`, or machine types.